packages/indexer/src/indexers/amm-market/amm-market-account-indexer.ts

```diff
@@ -0,0 +1,69 @@
+import { AccountInfoIndexer } from "../account-info-indexer";
+import { rpcReadClient } from "../../connection";
+import { AccountInfo, Context, PublicKey } from "@solana/web3.js";
+import { schema, usingDb } from "@metadaoproject/indexer-db";
+import { Err, Ok } from "../../match";
+import { BN } from "@coral-xyz/anchor";
+
+type TwapRecord = typeof schema.twaps._.inferInsert;
+
+export enum AmmAccountIndexerError {
+  GeneralError = "GeneralError",
+}
+
+// Doing this rather than class implements pattern due to
+// https://github.com/microsoft/TypeScript/issues/41399
+export const AmmMarketAccountUpdateIndexer: AccountInfoIndexer = {
+  index: async (
+    accountInfo: AccountInfo<Buffer>,
+    account: PublicKey,
+    context?: Context
+  ) => {
+    try {
+      const ammMarketAccount = await rpcReadClient.markets.amm.decodeMarket(
+        accountInfo
+      );
+      //index twap
+
+      // agg is 0 so skipping
+      if (ammMarketAccount.oracle.aggregator.toNumber() === 0)
+        return Ok({ acct: account.toString() });
+
+      const twapCalculation: BN = ammMarketAccount.oracle.aggregator.div(
+        ammMarketAccount.oracle.lastUpdatedSlot.sub(
+          ammMarketAccount.createdAtSlot
+        )
+      );
+      const twapNumber: number = twapCalculation.toNumber();
+      const newTwap: TwapRecord = {
+        curTwap: BigInt(twapNumber),
+        marketAcct: account.toString(),
+        observationAgg: ammMarketAccount.oracle.aggregator
+          .toNumber()
+          .toString(),
+        proposalAcct: ammMarketAccount.proposal.toString(),
+        // alternatively, we could pass in the context of the update here
+        updatedSlot: context
+          ? BigInt(context.slot)
+          : BigInt(ammMarketAccount.oracle.lastUpdatedSlot.toNumber()),
+        createdAt: new Date(),
+      };
+
+      const twapUpsertResult = await usingDb((db) =>
```

Comment:
The ONLY thing I'd say (since you're already getting the market stuffs) is to also store price...

Reply:
ahhh okay, from the price property in the observation?

Reply:
Yea so it's kinda clunky (and likely merits it's own function) but if you fetch account data, mind as well update what you can from it since you already expended the resource to get it.

Reply:
Am I adding a new column n the twaps table, or is the price here like the spot price of the market or something?

Reply:
Cool added to TODO